File: 程式語言 V2答案檔/Python/310.py
def compute(n):
    A_list = []
    for i in range(1, n):
        s = str(i)
        k = len(s)
        cnt = 0
        for j in range(k):
            cnt += int(s[j]) ** k
        if cnt == i:
            A_list.append(i)
    return A_list

n = int(input())
A_list = compute(n)
sum = 0
for x in A_list:
    print(x)
    sum += x
print(sum)

File: 程式語言 V2答案檔/PHP/310.php
<?php
    function sub_compute($n){
        // The function to compute the Armstrong number.
        $str = strval($n);
        $total = 0;
        for($i = 0; $i < strlen($str); $i++){
            $total += intval($str[$i]) ** strlen($str);
        }
        return $total;
    }
    function compute($n){
        $array = array();
        for($i = 1; $i <= $n; $i++){
            if($i == sub_compute($i)){
                array_push($array, $i);
            }
        }
        return $array;
    }
    $n = intval(fgets(STDIN));
    $array = compute($n);
    $sum = 0;
    for($i = 0; $i < count($array); $i++){
        $sum += $array[$i];
        printf("%d\n", $array[$i]);
    }
    printf("%d\n", $sum);
?>